rewrite-maven/src/main/java/org/openrewrite/maven/UpgradeDependencyVersion.java

```diff
@@ -125,35 +126,79 @@ public String getDescription() {
+                        if (newerVersion != null) {
+                            Optional<Xml.Tag> version = tag.getChild("version");
+                            if (version.isPresent()) {
+                                String requestedVersion = d.getRequested().getVersion();
+                                if (requestedVersion != null && requestedVersion.startsWith("${") && !implicitlyDefinedVersionProperties.contains(requestedVersion)) {
+                                    String propertyName = requestedVersion.substring(2, requestedVersion.length() - 1);
+                                    if (!getResolutionResult().getPom().getRequested().getProperties().containsKey(propertyName)) {
+                                        getPomDeclaringProperty(getResolutionResult(), propertyName)
+                                                .ifPresent(pom -> accumulator.pomProperties.add(
+                                                        new PomProperty(requireNonNull(pom.getSourcePath()), propertyName)));
+                                    }
+                                }
+                            }
+                        }
+                    } catch (MavenDownloadingException e) {
+                        return e.warn(tag);
+                    }
+                }
+                return super.visitTag(tag, ctx);
             }
+
         };
     }
 
     @Override
-    public TreeVisitor<?, ExecutionContext> getVisitor(Set<GroupArtifact> projectArtifacts) {
-        VersionComparator versionComparator = Semver.validate(newVersion, versionPattern).getValue();
-        assert versionComparator != null;
-
+    public TreeVisitor<?, ExecutionContext> getVisitor(Accumulator accumulator) {
         return new MavenIsoVisitor<ExecutionContext>() {
             private final XPathMatcher PROJECT_MATCHER = new XPathMatcher("/project");
+            private final VersionComparator versionComparator =
+                    requireNonNull(Semver.validate(newVersion, versionPattern).getValue());
 
             @Override
             public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
                 Xml.Tag t = super.visitTag(tag, ctx);
                 try {
-                    if (isDependencyTag(groupId, artifactId)) {
+                    if (isPropertyTag()) {
+                        if (!tag.getValue().map(newVersion::equals).orElse(false)) {
+                            Path pomSourcePath = getResolutionResult().getPom().getRequested().getSourcePath();
+                            for (PomProperty pomProperty : accumulator.pomProperties) {
+                                if (pomProperty.pomFilePath.equals(pomSourcePath) &&
+                                    pomProperty.property.equals(tag.getName())) {
+                                    doAfterVisit(new ChangeTagValueVisitor<>(tag, newVersion));
```

Comment:
Here we look to set the property value to `newVersion`, but `newVersion` can contain a [dependency version selector](https://docs.openrewrite.org/reference/dependency-version-selectors) pattern (e.g. `10.15.x`) if I'm not mistaken, as opposed to a literal version `10.15.0` as seen in the tests. We likely need to handle such cases as well, by determining the actual version value to fill in here. (unless I missed something, in which case please correct me!)

Reply:
Excellent catch! Handled this in the next version.